src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,275 @@
+import { RequestSender, Response } from '@bigcommerce/request-sender';
+
+import {
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethod,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions,
+    PaymentStrategyActionCreator,
+} from '../..';
+import { CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { toFormUrlEncoded } from '../../../common/http-request';
+import { default as bind } from '../../../common/utility/bind-decorator';
+import { OrderActionCreator, OrderPaymentRequestBody, OrderRequestBody } from '../../../order';
+import { ChasePayScriptLoader, ChasePaySuccessPayload } from '../../../payment/strategies/chasepay';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { CryptogramInstrument } from '../../payment';
+import PaymentStrategy from '../payment-strategy';
+import WepayRiskClient from '../wepay/wepay-risk-client';
+
+export default class ChasepayPaymentStrategy extends PaymentStrategy {
+    private _button?: HTMLElement;
+    private _logoContainer?: string;
+    private _methodId?: string;
+    private _paymentMethod?: PaymentMethod;
+    private _storeLanguage?: string;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _chasePayScriptLoader: ChasePayScriptLoader,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _requestSender: RequestSender,
+        private _wepayRiskClient: WepayRiskClient,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _checkoutActionCreator: CheckoutActionCreator
+
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        if (!options.chasepay) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.chasepayCheckoutOptions" argument is not provided.');
+        }
+
+        const {
+            onPaymentSelect = () => {},
+            logoContainer,
+            walletButton,
+        } = options.chasepay;
+
+        const storeConfig = this._store.getState().config.getStoreConfig();
+
+        if (!storeConfig) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        this._storeLanguage = storeConfig.storeProfile.storeLanguage;
+        this._logoContainer = logoContainer;
+        this._methodId = methodId;
+
+        if (options.gatewayId === 'wepay') {
+            this._wepayRiskClient.initialize();
+        }
+
+        this._button = walletButton && document.getElementById(walletButton) || undefined;
```

Comment:
We use the || undefined in case that the `walletButton` is set but it does not exist on the DOM, since `document.getElementById(walletButton)` returns null if the element does not exist. So it is necessary.